inst/include/sampler.h

```diff
@@ -641,6 +725,8 @@ namespace smc {
         }
         else {
             nResampled = 0;
+            uRSIndices = arma::zeros<arma::Col<unsigned int> >(static_cast<int>(N));

```

Comment:
I guess we could do this conditionally on wanting to store this information, but I can see the case for keeping things simple.

Reply:
At first, an `if()` looked a bit ugly to me, so I deleted it for the moment. 

Now I believe, an external person might get confused by this redundancy in case of `HistoryType::RAM` so I added it now.

Reply:
I don't have strong views, it just seems like an unnecessary operation if the information isn't wanted and I still remember when computers were slow...